tests/phpunit/TestFramework/PhpUnit/Adapter/PhpUnitAdapterTest.php

```diff
@@ -35,7 +35,9 @@
 
 namespace Infection\Tests\TestFramework\PhpUnit\Adapter;
 
-use Infection\AbstractTestFramework\MemoryUsageAware;
+use function array_map;
+use Generator;
+use Infection\PhpParser\Visitor\IgnoreNode\PhpUnitCodeCoverageAnnotationIgnorer;
```

Comment:
I'm not very happy with this test:

- It's a pure unit test for a third-party adapter, IMO it should warrant an integration test (either besides this one or instead of this one)
- We do not test anything coming from the abstract adapter which is a bit of a miss IMO

So I'm thinking of making this an integration test and capture the behaviour inherited from the abstract adapter. Do you have any objection or something that I missed?

Reply:
TBH, I didn't get you. We have tens of e2e tests for PHPUnit adapter. And here is a unit tests. What's wrong with it?

Reply:
I feel it would be better to have an integration test for the `PhpUnitAdapter` where you can test all the above - that's fine, but also the following:

- get the version
- get the command lines

Sure an e2e test eventually guarantees that overall it's working, but it's also nice to be able to capture some of that behaviour in a more isolated environment and having a strictly unit test kind of prevents that